### ISSUE TITLE
Add a way to "close"/"disable" folder temporarily without removing it from workspace (Eclipse style) 

### ISSUE THREAD

**Issue by Ivanca** [microsoft/vscode#205160]
## Scenario
I have only one workspace. In this workspace I have about 15 folders (1 folder == 1 git repo) that are all related projects: folder1 ... folder15
Sometimes I want to do some work that only touches a few of those folders. However, when I search, I get matches from all 15 folders.
I want to disable folder12 (to exclude it from searches etc.), but without removing it from workspace. I.e. the folder would be effectively dead, but would still show up on the list of folders, so it can easily be reenabled.
Right now AFAIU, I can only remove project from workspace when I right click on it, but then I have to search for it again on disk when I want to readd it.

## Proposal
There should be a new option: Disable folder / Enable folder in the menu below:

![image](https://user-images.githubusercontent.com/1437027/69337295-c9c8df80-0c60-11ea-9511-64234bfa0503.png)

Related screeshots from Eclipse:

![image](https://user-images.githubusercontent.com/1437027/69337099-5aeb8680-0c60-11ea-89ca-b2da660143da.png)
![image](https://user-images.githubusercontent.com/1437027/69337149-79ea1880-0c60-11ea-9240-ea6ee6344234.png)

More explanation
My use case is not about resource consumption, but about:

avoiding search matches that are useless in CTRL-SHIFT-F / CTRL-SHIFT-H (workspace-wide find/replace)
avoiding file matches that are useless in CTRL-P (quick file open)
Assume you work on foobar but you have one annoying folder which has 1000 matches of foobar which are not relevant to your current work. You want to disable it for a day or two, or more, but not completely thrash it out from workspace.

Alternatives
As an alternative, to prevent search matches, one could use "files to exclude" option, but this is just a small textbox, and not great from UX point of view (feels like editing PATH in Windows XP).

There's no easy way to prevent matches in quick file open though that I am aware of.

**Comment by gjsjohnmurray** [comment]
Did you try using the `Find in Folder...` option from the context menu of one of your workspace root folders as a quick way of populating the "Files to include" field (I moved Search view to secondary sidebar for the screenshot below)?

![image](https://github.com/microsoft/vscode/assets/6726799/74561acb-744d-4ffd-a02c-094a0f58a403)

**Comment by Ivanca** [comment]
Yes, my problem is also performance wise, I have a single workspace with many Java repositories and also Angular repositories, which is pretty handy for a lot of operations, but it makes my computer really slow, despite being a "gamer laptop", I want to be able to disable some of those at any given time so they are completely ignored by Java plugins (that always tries to index, find maven scripts, etc)

**Comment by Ivanca** [comment]
Another helpful thing this feature would do is that when I use ctrl+p to select a file I would know that its only looking for the folders I have enabled at the time.

**Comment by andreamah** [comment]
If you would not want to see those folders in search, you can add the path to `search.exclude` in settings. Does that help?

**Comment by Ivanca** [comment]
> If you would not want to see those folders in search, you can add the path to `search.exclude` in settings. Does that help?

It does not, to be user-friendly it would be able to right-click and exclude them, also I want to exclude them from being seen by plugins (e.g java plugins when working on java projects, react plugins when working with react folders)

**Comment by andreamah** [comment]
In terms of being seen by java plugins and such, that would be something best managed by the Java Extension (so you should probably put it in their issue tracker). I believe that what you're asking for on our side, however, is for a menu item to hide the folder from showing in vscode. Therefore, this would be a right-click menu item that would add that folder to the list of paths under `files.exclude` in the settings (this also will remove it from the search). Is this what you're asking for?

**Comment by Ivanca** [comment]
Not at all, a lot of plugins create their own menu, for example npm-scripts create an entry for each project where the commands inside package.json are exposed in the UI, asking the authors of each plugin that create a menu of the folders to implement this functionality its unreasonable, VSCode should have a way to hide them and make them fully invisible to plugins as well as search, ctrl+p, debugging/start panel, et al

**Comment by andreamah** [comment]
@lramos15 I believe this is closer to your area, feel free to re-assign.

**Comment by Ivanca** [comment]
I probably can make a plugin for this functionality because I understand that maybe there is not enough people who want this; but I don't think the existing API allows for such functionality, or am I wrong? @lramos15

**Comment by Ivanca** [comment]
ok, thanks for nothing.

**Comment by arthar360** [comment]
This would a useful feature. Using search restriction is not always convenient. Currently I open new windows with specific repos. It too dirty solution. Adding disable folder/project will help a lot.

**Comment by SephiRok** [comment]
Also looking for this functionality. That is, disable & enable a whole folder and all associated functionality (search, source control, etc.) with one or two clicks.

**Comment by daniele-orlando** [comment]
Looking for this too. I have monorepo with ~10 apps and I have 1 VSCode project. Most of the time I work across all 10 apps syncing data models and shared stuff. But other times I want to focus on only 2 apps, so I want to search and fast open files only from those 2 apps. The combo of apps is not always the same,  sometimes I have 3 apps and they are not always the same. 
It is not feasible to have one VSCode project for every combination of the matrix, nor to exclude manually the folders every time.

I really would to have the possibility to disable/enable open folders multi-selecting them.

**Comment by Ivanca** [comment]
### This feature would be **a perfect fit for AI**...

Because by default the models would only use the active files as context, and we know how important is that, don't we Mr Microsoft?

**Comment by psychowood** [comment]
Just in case someone is still looking for a solution: https://github.com/psychowood/workspace-folder-disabler
